File: conection/lib/models/detail.dart
import 'typeofnew.dart';
class detail{
  int id;
  int numberBath;
  int numberLiving;
  String director;
  typeofnew Typeofnew;
  int typeofnewId;
  String liveType;
  int liveTypeId;
  int motelId;

  detail(int id,int numberBath,int numberLiving,String director
      ,typeofnew Typeofnew,int typeofnewId,String liveType,
      int liveTypeId, int motelId ){
    this.id=id;
    this.numberBath=numberBath;
    this.director=director;
    this.Typeofnew =Typeofnew;
    this.typeofnewId=typeofnewId;
    this.liveType=liveType;
    this.liveTypeId=liveTypeId;
    this.motelId=motelId;
  }
  detail.fromJson(Map json)
      : id = json['id'],
        numberBath = json['numberBath'],
        director = json['director'],
        Typeofnew = json['Typeofnew'],
        typeofnewId = json['typeofnewId'],
        liveType = json['liveType'],
        liveTypeId = json['liveTypeId'],
        motelId = json['motelId'];
}

File: Sourcode/back/Websitedangtintimkiemnhatro/Websitedangtintimkiemnhatro/Models/Detail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Websitedangtintimkiemnhatro.Models
{
    public class Detail
    {
        [Key]
        public int Id { get; set; }
        public int NumberBath { get; set; }
        public int NumberLiving { get; set; }
        public string Director { get; set; }
        public Typeofnew Typeofnew { get; set; }
        [ForeignKey("TypeofnewId")]
        [Required]
        public int TypeofnewId { get; set; }
        public LiveType LiveType { get; set; }
        [ForeignKey("LiveTypeId")]
        [Required]
        public int LiveTypeId { get; set; }
        public Motel Motel { get; set; }
        [ForeignKey("MotelId")]
        [Required]
        public int MotelId { get; set; }
    }
}
